tests/unit/s2n_connection_preferences_test.c

```diff
@@ -243,11 +243,18 @@ int main(int argc, char **argv)
         EXPECT_FAILURE(s2n_connection_get_signature_preferences(conn, &signature_preferences));
         EXPECT_FAILURE(s2n_connection_get_ecc_preferences(conn, &ecc_preferences));
 
+        DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(),
+                s2n_config_ptr_free);
+        EXPECT_NOT_NULL(config);
+        EXPECT_SUCCESS(s2n_config_set_cipher_preferences(config, "20240501"));
+
```

Comment:
Setting the cipher preferences to a specific security policy might be confusing given that you then immediately manually set the policy to NULL on line 257.

But I guess you're doing this pre-emptively as part of removing implicit defaults? It might actually be preferable to let whatever mechanism you're working on for that catch this. I know we discussed comments indicating that the call to set cipher preferences was added automatically, which would clarify this logic.

Reply:
Yea its a bit odd in this test due to the immediate NULL-ing. Ill do as you suggest, remove it and fix it during the implicit detection.